lib/utils.js

```diff
@@ -842,3 +855,32 @@ function idsHaveDuplicates(ids) {
 function isClass(fn) {
   return fn && fn.toString().startsWith('class ');
 }
+
+/**
+ * Accept an element, and attach the __parent property to it, unless no object given, while also
+ * making sure to check for already created properties
+ *
+ * @param {object} element
+ * @param {Model} parent
+ */
+function applyParentProperty(element, parent) {
+  assert.strictEqual(typeof element, 'object', 'Non object element given to assign parent');
+  const {constructor: {modelBuilder: {settings: builderSettings} = {}} = {}} = element;
+  if (!builderSettings || !builderSettings[BUILDER_PARENT_SETTING]) return; // parentRef flag not enabled on ModelBuilder settings
+  if (element.hasOwnProperty(PARENT_PROPERTY_NAME)) { // property already created on model, just assign
+    const existingParent = element[PARENT_PROPERTY_NAME];
+    if (existingParent && existingParent !== parent) { // parent re-assigned (child model assigned to other model instance)
+      g.warn('Re-assigning child model instance to an other parent than the original! ' +
+        'Although supported, this is not a recommended practice: ' +
+        `${element.constructor.name} -> ${parent.constructor.name}`);
```

Comment:
Good error messages are actionable. Please explain the user what is the recommended practice and/or point them to a place in our docs where they can find information on how to resolve this warning.

Reply:
Added recommendation to warning message